assessments/projects/fluentd/self-assessment.md

```diff
@@ -0,0 +1,223 @@
+- **Parser**: Sometimes, the <parse> directive for input plugins (e.g. in_tail, in_syslog, in_tcp and in_udp) cannot parse the user's custom data format (for example, a context-dependent grammar that can't be parsed with a regular expression). To address such cases, Fluentd has a pluggable system that enables the user to create their own parser formats.
+- **Filter**: Filter plugins enable Fluentd to modify event streams, such as filtering out events by grepping the value of one or more fields, enriching events by adding new fields, and deleting or masking certain fields for privacy and compliance.
+- **Output**: Output plugins have 3 buffering and flushing modes. Non-Buffered mode does not buffer data and write out results immediately. Synchronous Buffered mode has "staged" buffer chunks (a chunk is a collection of events) and a queue of chunks, and its behavior can be controlled by <buffer> section. Asynchronous Buffered mode also has "stage" and "queue", but the output plugin will not commit writing chunks in methods synchronously, but commit them later.
+- **Formatter**: If the output format for an output plugin does not meet one's needs, Fluentd has a pluggable system called Formatter that lets the user extend and re-use custom output formats.
+- **Storage**: When the input/filter/output plugin needs to save its internal state in memory, storage, or key-value store, Fluentd has a pluggable system called Storage that lets a plugin store and reuse its internal state as key-value pairs.
+- **Service Discovery**: Some plugins support <service_discovery> (e.g. out_forward). Sometimes, the service discovery for an output plugin does not meet one's needs. Fluentd has a pluggable system called Service Discovery that lets the user extend and reuse custom output service discovery.
+- **Buffer**: Buffer plugins are used by output plugins. For example, out_s3 uses buf_file by default to store incoming stream temporally before transmitting to S3.
+- **Metrics**: Sometimes, the input/filter/output plugin needs to save its internal metrics in memory, influxdb or prometheus format ready in instances. Fluentd has a pluggable system called Metrics that lets a plugin store and reuse its internal state as metrics instances.
+
+Fluentd is equipped with a password-based authentication mechanism, which verifies the identity of each client using a shared secret key. The secure_forward plugins for input and output (in_secure_forward and out_secure_forward) accepts and sends messages via SSL with authentication.
+
+A few popular use cases for Fluentd include:
+1. Integration with reading and writing from Kafka
+2. Ingesting long term data into long term data stores (E.g. Hadoop, Google Cloud storage, Amazon S3)
+3. Migrating from proprietary backends by routing data to multiple destinations
+4. Listening on syslog for firewalls and networks devices and performing GeoIP or other enrichment
+5. Routing application logs from apps running in Kubernetes
```

Comment:
Please double-check... this doesn't seem like it should live within the Actors section.